pro_tes/middleware/task_distribution.py

```diff
@@ -2,13 +2,22 @@
 
 from copy import deepcopy
 import random
-from typing import List, Optional
+from urllib.parse import urlparse
+from socket import gaierror, gethostbyname
+from typing import Dict, List, Set, Tuple, Optional
+from itertools import combinations
+from ip2geotools.errors import InvalidRequestError
+from ip2geotools.databases.noncommercial import DbIpCity
+from geopy.distance import geodesic
```

Comment:
Follow the import rules from PEP8 and use alphabetic sorting as described above. Check the other modules you changed as well.